workshop/Lab3/README.md

```diff
@@ -0,0 +1,189 @@
+# Lab 3 Deploy an application with a secret containing creds to a backing store
+
+Learn to create a Kubernetes secret containing the credentials to connect to a service,
+and attach that secret to an application to consume that service.
+
+# Create a secret
+
+[Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) are a Kubernetes resource
+intended to hold sensitive information such as usernames, passwords, or tokens. While your 
+application may need this information to function, it is common that you wish to restrict
+access to this information. Previously, we were hardcoding our sensitive info directly into
+our deployment declaration, where it was readable by anyone with access to our deployment.
+
+To create a secret, encode the values you wish it to contain in base 64, and then add them to
+the secrets.yaml file, like so:
+
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: guestbookcreds
+type: Opaque
+data:
+  host: INSERT_HOST_HERE
+  port: INSERT_PORT_HERE
+  password: INSERT_PASSWORD_HERE
+```
+
+To encode a value in base 64, use the base64 terminal command:
+
+```
+echo -n "INSERT_HOST_HERE" | base 64
+```
+
+This should output a long string such as
+cHViLXJlZGlzLTE3NjgxLmRhbC0wNS4xLnNsLmdhcmFudGlhZGF0YS5jb20K. Note the -n
+flag for echo so the resulting string doesn't contain a newline character.
+Insert these values into the relevant fields, and then create the secret:
+
+```
+kubectl create -f secret.yaml
+```
+
+You should be able to see it by getting secrets:
+
+```
```

Comment:
can add a `console` annotation for nicer formatting.

Reply:
not sure what you mean by this

Reply:
added in relevant sections